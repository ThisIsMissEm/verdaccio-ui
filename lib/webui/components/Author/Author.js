"use strict";
/**
 * @prettier
 */
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const index_1 = __importDefault(require("@material-ui/core/Avatar/index"));
const index_2 = __importDefault(require("@material-ui/core/List/index"));
const index_3 = __importDefault(require("@material-ui/core/ListItemText/index"));
const Version_1 = require("../../pages/version/Version");
const styles_1 = require("./styles");
class Authors extends react_1.Component {
    constructor() {
        super(...arguments);
        this.renderAuthor = ({ packageMeta }) => {
            const { author, name: packageName, version } = packageMeta.latest;
            if (!author) {
                return null;
            }
            const avatarComponent = react_1.default.createElement(index_1.default, { alt: author.name, src: author.avatar });
            return (react_1.default.createElement(index_2.default, { subheader: react_1.default.createElement(styles_1.Heading, { variant: 'subheading' }, 'Author') },
                react_1.default.createElement(styles_1.AuthorListItem, null,
                    this.renderLinkForMail(author.email, avatarComponent, packageName, version),
                    react_1.default.createElement(index_3.default, { primary: author.name }))));
        };
    }
    render() {
        return (react_1.default.createElement(Version_1.DetailContextConsumer, null, (context) => {
            return this.renderAuthor(context);
        }));
    }
    renderLinkForMail(email, avatarComponent, packageName, version) {
        if (!email) {
            return avatarComponent;
        }
        return (react_1.default.createElement("a", { href: `mailto:${email}?subject=${packageName}@${version}`, target: '_top' }, avatarComponent));
    }
}
exports.default = Authors;
