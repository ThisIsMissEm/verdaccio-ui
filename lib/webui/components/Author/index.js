"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Author_1 = require("./Author");
exports.default = Author_1.default;
