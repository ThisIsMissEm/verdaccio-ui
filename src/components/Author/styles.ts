import styled from 'react-emotion';
import Typography from '@material-ui/core/Typography';

import ListItem from '../ListItem';

export const Heading = styled(Typography)`
  && {
    font-weight: 700;
    text-transform: capitalize;
  }
`;
export const AuthorListItem = styled(ListItem)`
  && {
    padding-left: 0;
    padding-right: 0;
  }
`;
