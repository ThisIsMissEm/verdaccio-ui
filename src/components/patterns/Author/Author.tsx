/**
 * @prettier
 */

/* eslint no-unused-vars: 0 */

import React from 'react';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';

import { DetailContext, DetailContextProps } from '../../../pages/Version';

interface Props {
  name: string;
  avatar?: string;
}

const Author: React.FC = () => {
  const renderAuthor: React.FC<DetailContextProps> = ({
    packageMeta: {
      latest: { author },
    },
  }) =>
    author ? (
      <Card>
        <CardContent>{renderAvatar(author)}</CardContent>
      </Card>
    ) : null;

  const renderAvatar: React.FC<Props> = ({ name, avatar }) => {
    return (
      <>
        <Avatar aria-label={name} src={avatar} />
        <Typography color={'textPrimary'} gutterBottom={true} variant={'caption'}>
          {name}
        </Typography>
      </>
    );
  };

  return <DetailContext.Consumer>{context => context && renderAuthor(context)}</DetailContext.Consumer>;
};

export default Author;
