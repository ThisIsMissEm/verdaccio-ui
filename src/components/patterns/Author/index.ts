/**
 * @prettier
 */

export { default } from './Author';
