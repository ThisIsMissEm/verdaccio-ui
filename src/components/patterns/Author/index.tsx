/* eslint no-unused-vars: 0 */

import React from 'react';

import { DetailContextConsumer } from '../../pages/version';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CopyToClipBoard from '../CopyToClipBoard';
import CardHeader from '@material-ui/core/CardHeader';
import Avatar from '@material-ui/core/Avatar';
import CardActions from '@material-ui/core/CardActions';
import Typography from '@material-ui/core/Typography';

interface Props {
  name: string
  email: string
  url: string
  avatar: string
}

const Authors: React.FC<Props> = ({ name, email, url, avatar }) => {

  const renderAvatar = ({name, email, url, avatar}: Avatar) => (
    <>
      <Avatar aria-label={name} src={avatar} />
      <Typography color={'textPrimary'} gutterBottom={true} variant={'caption'}>
        {name}
      </Typography>
    </>
  )

  const renderAuthor = ({ packageMeta }) => {
    const { author } = packageMeta.latest;

    if (!author) {
      return null;
    }

    return (
      <Card>
        <CardContent>{renderAvatar(author)}</CardContent>
      </Card>
    );
  };

  return (
    <DetailContextConsumer>
      {(context: any) => {
        return renderAuthor(context);
      }}
    </DetailContextConsumer>
  );
}

export default Authors

// class Authors extends React.Component<any, any> {
//   render() {
//     return (
//       <DetailContextConsumer>
//         {(context) => {
//           return this.renderAuthor(context);
//         }}
//       </DetailContextConsumer>
//     );
//   }

//   renderAuthor = ({packageMeta}) => {
//     const {author} = packageMeta.latest;

//     if (!author) {
//       return null;
//     }

//     return (
//       <Card>
//         <CardContent>{this.renderAvatar(author)}</CardContent>
//       </Card>
//     );
//   };

//   renderAvatar = ({name, email, url, avatar}: Avatar) => {
//     return (
//       <>
//         <Avatar aria-label={name} src={avatar} />
//         <Typography color={'textPrimary'} gutterBottom={true} variant={'caption'}>
//           {name}
//         </Typography>
//       </>
//     );
//   };
// }

// export default Authors;
