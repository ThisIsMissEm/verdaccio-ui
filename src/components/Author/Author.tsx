import React, { Component, ReactNode } from 'react';

import Avatar from '@material-ui/core/Avatar';
import List from '@material-ui/core/List';
import ListItemText from '@material-ui/core/ListItemText';

import { DetailContextConsumer } from '../../pages/version/Version';
import { Heading, AuthorListItem } from './styles';
import { isEmail } from '../../utils/url';

class Authors extends Component<any, any> {
  render() {
    return (
      <DetailContextConsumer>
        {(context: any) => {
          return context && context.packageMeta && this.renderAuthor(context.packageMeta);
        }}
      </DetailContextConsumer>
    );
  }

  renderLinkForMail(email: string, avatarComponent: ReactNode, packageName: string, version: string) {
    if (!email || isEmail(email) === false) {
      return avatarComponent;
    }

    return (
      <a href={`mailto:${email}?subject=${packageName}@${version}`} target={'_top'}>
        {avatarComponent}
      </a>
    );
  }

  renderAuthor = packageMeta => {
    const { author, name: packageName, version } = packageMeta.latest;

    if (!author) {
      return null;
    }

    const avatarComponent = <Avatar alt={author.name} src={author.avatar} />;
    return (
      <List subheader={<Heading variant={'subheading'}>{'Author'}</Heading>}>
        <AuthorListItem>
          {this.renderLinkForMail(author.email, avatarComponent, packageName, version)}
          <ListItemText primary={author.name} />
        </AuthorListItem>
      </List>
    );
  };
}

export default Authors;
