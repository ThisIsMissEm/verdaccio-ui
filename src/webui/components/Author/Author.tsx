/**
 * @prettier
 */

import React, { Component, ReactNode } from 'react';

import Avatar from '@material-ui/core/Avatar/index';
import List from '@material-ui/core/List/index';
import ListItemText from '@material-ui/core/ListItemText/index';

import { DetailContextConsumer } from '../../pages/version/Version';
import { Heading, AuthorListItem } from './styles';
import { isEmail } from '../../utils/url';

class Authors extends Component<any, any> {
  render() {
    return (
      <DetailContextConsumer>
        {(context: any) => {
          return this.renderAuthor(context);
        }}
      </DetailContextConsumer>
    );
  }

  renderLinkForMail(email: string, avatarComponent: ReactNode, packageName: string, version: string) {
    if (!email || isEmail(email) === false) {
      return avatarComponent;
    }

    return (
      <a href={`mailto:${email}?subject=${packageName}@${version}`} target={'_top'}>
        {avatarComponent}
      </a>
    );
  }

  renderAuthor = ({ packageMeta }: any) => {
    const { author, name: packageName, version } = packageMeta.latest;

    if (!author) {
      return null;
    }

    const avatarComponent = <Avatar alt={author.name} src={author.avatar} />;
    return (
      <List subheader={<Heading variant={'subheading'}>{'Author'}</Heading>}>
        <AuthorListItem>
          {this.renderLinkForMail(author.email, avatarComponent, packageName, version)}
          <ListItemText primary={author.name} />
        </AuthorListItem>
      </List>
    );
  };
}

export default Authors;
