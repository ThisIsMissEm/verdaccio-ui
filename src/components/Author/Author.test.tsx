import React from 'react';
import { shallow } from 'enzyme';

import Author from './Author';

describe('<Author /> component', () => {
  const props = {
    name: 'austria',
  };
  test('should render the component in default state', () => {
    const wrapper = shallow(<Author />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
