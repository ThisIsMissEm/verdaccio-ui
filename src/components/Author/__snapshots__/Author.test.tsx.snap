// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Author /> component should render the component when there is no author email 1`] = `
Authors {
  "context": Object {},
  "props": Object {},
  "refs": Object {},
  "renderAuthor": [Function],
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <Authors />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <DetailContextConsumer>
        [Function]
      </DetailContextConsumer>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
